spec/unit/sibling_decorator_spec.rb

```diff
@@ -0,0 +1,270 @@
+
+    describe "#previous_sibling" do
+      it "should be nil" do
+        expect(decorator.previous_sibling).to be_nil
+      end
+    end
+
+    describe "#next_sibling" do
+      it "should be present" do
+        expect(decorator.next_sibling.section_id).to eq("next-sibling")
+        expect(decorator.next_sibling.base_path).to eq("/guidance/a-manual/next-sibling")
+        expect(decorator.next_sibling.title).to eq("Next sibling title")
+      end
+    end
+  end
+
+  context "for a section that is a last child" do
+    let(:child_section_groups) {
+      [
+        double(
+          child_sections: [
+            double(
+              section_id: "previous-sibling",
+              base_path: "/guidance/a-manual/previous-sibling",
+              title: "Previous sibling title",
+            ),
+            double(
+              section_id: document_id,
+              base_path: document_base_path,
+              title: "Child section title",
+            ),
+          ]
+        ),
+      ]
+    }
+
+    describe "#previous_sibling" do
+      it "should be present" do
+        expect(decorator.previous_sibling.section_id).to eq("previous-sibling")
+        expect(decorator.previous_sibling.base_path).to eq("/guidance/a-manual/previous-sibling")
+        expect(decorator.previous_sibling.title).to eq("Previous sibling title")
+      end
+    end
+
+    describe "#next_sibling" do
+      it "should be nil" do
+        expect(decorator.next_sibling).to be_nil
+      end
+    end
+  end
+
+  context "for a section that is a mid child" do
+    let(:child_section_groups) {
+      [
+        double(
+          child_sections: [
+            double(
+              section_id: "previous-sibling",
+              base_path: "/guidance/a-manual/previous-sibling",
+              title: "Previous sibling title",
+            ),
+            double(
+              section_id: document_id,
+              base_path: document_base_path,
+              title: "Child section title",
+            ),
+            double(
+              section_id: "next-sibling",
+              base_path: "/guidance/a-manual/next-sibling",
+              title: "Next sibling title",
+            ),
+          ]
+        ),
+      ]
+    }
+
+    describe "#previous_sibling" do
+      it "should be present" do
+        expect(decorator.previous_sibling.section_id).to eq("previous-sibling")
+        expect(decorator.previous_sibling.base_path).to eq("/guidance/a-manual/previous-sibling")
+        expect(decorator.previous_sibling.title).to eq("Previous sibling title")
+      end
+    end
+
+    describe "#next_sibling" do
+      it "should be present" do
+        expect(decorator.next_sibling.section_id).to eq("next-sibling")
+        expect(decorator.next_sibling.base_path).to eq("/guidance/a-manual/next-sibling")
+        expect(decorator.next_sibling.title).to eq("Next sibling title")
+      end
+    end
+
+    context "for a section not recognised by its parent a child" do
```

Comment:
was supposed to be 'as a child', but I can just remove it. Thanks for spotting it!